src/devices/Adxl357/Adxl357.cs

```diff
@@ -0,0 +1,190 @@
+        }
+
+        /// <summary>
+        /// Gets or sets the sensitivity of the accelerometer.
+        /// </summary>
+        public AccelerometerRange AccelerometerRange
+        {
+            get => (AccelerometerRange)ReadByte(Register.SET_RANGE_REG_ADDR);
+            set
+            {
+                var currentValue = ReadByte(Register.SET_RANGE_REG_ADDR);
+                var newValue = currentValue | (byte)value;
+                WriteRegister(Register.SET_RANGE_REG_ADDR, (byte)newValue);
+            }
+        }
+
+        private double GetTemperature()
+        {
+            Span<byte> data = stackalloc byte[2] { 0, 0 };
+
+            ReadBytes(Register.TEMPERATURE_REG_ADDR, data);
+
+            double value = BinaryPrimitives.ReadUInt16BigEndian(data);
+
+            return 25 + (value - 1852) / -9.05;
+        }
+
+        private void Reset()
+        {
+            WriteRegister(Register.RESET_REG_ADDR, 0x52);
+            Thread.Sleep(100);
+        }
+
+        private void PowerOn()
+        {
+            WriteRegister(Register.POWER_CTR_REG_ADDR, 0x00);
+            Thread.Sleep(100);
+        }
+
+        private Vector3 GetRawAccelerometer()
+        {
+            Span<byte> data = stackalloc byte[9] { 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+
+            var ace = new Vector3();
+
+            if (CheckDataReady())
+            {
+                ReadBytes(Register.FIFO_DATA_REG_ADDR, data);
+
+                ace.X = GetValueForOneAxis(data[0], data[1], data[2]);
+                ace.Y = GetValueForOneAxis(data[3], data[4], data[5]);
+                ace.Z = GetValueForOneAxis(data[6], data[7], data[8]);
+            }
+
+            return ace;
+        }
+
+        private float GetValueForOneAxis(byte firstByte, byte secondByte, byte thirdByte)
+        {
+            int value = (firstByte << 12) | (secondByte << 4) | (thirdByte >> 4);
+
+            value = (value & 0x80000) == 0x80000 ? -value : value;
+
+            return value * _factory;
+        }
+
+        private bool CheckDataReady()
+        {
+            var status = GetAdxl357Status();
+            return (status & 0x01) == 0x01;
+        }
+
+        private byte GetAdxl357Status() => ReadByte(Register.STATUS_REG_ADDR);
+
+        private void WriteRegister(Register register, byte data)
+        {
+            Span<byte> dataout = stackalloc byte[]
+            {
+                (byte)register, data
+            };
+
+            _i2CDevice.Write(dataout);
+        }
+
+        private byte ReadByte(Register register)
+        {
+            _i2CDevice.WriteByte((byte)register);
+            return _i2CDevice.ReadByte();
+        }
+
+        private void ReadBytes(Register register, Span<byte> readBytes)
+        {
+            _i2CDevice.WriteByte((byte)register);
+            _i2CDevice.Read(readBytes);
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            _i2CDevice.Dispose();
```

Comment:
```suggestion
            _i2CDevice?.Dispose();
```

Reply:
Since nullable reference types are enabled globally, this instance is known not to be null, isn't it?